Library/Formula/libgee.rb

```diff
@@ -16,10 +16,63 @@ class Libgee < Formula
   depends_on "vala" => :build
   depends_on "gobject-introspection"
 
+  # ensures that the gobject-introspection files remain within the keg
+  patch :DATA
+
   def install
     system "./configure", "--prefix=#{prefix}",
-                          "--disable-dependency-tracking",
-                          "--enable-introspection=yes"
+                          "--disable-dependency-tracking"
     system "make", "install"
   end
+
+  test do
+    (testpath/"test.c").write <<-EOS.undent
+      #include <gee.h>
+
+      int main(int argc, char *argv[]) {
+        GType type = gee_traversable_stream_get_type();
+        return 0;
+      }
+    EOS
+    gettext = Formula["gettext"]
+    glib = Formula["glib"]
+    flags = (ENV.cflags || "").split + (ENV.cppflags || "").split + (ENV.ldflags || "").split
+    flags += %W[
+      -I#{gettext.opt_include}
+      -I#{glib.opt_include}/glib-2.0
+      -I#{glib.opt_lib}/glib-2.0/include
+      -I#{include}/gee-0.8
+      -L#{gettext.opt_lib}
+      -L#{glib.opt_lib}
+      -L#{lib}
+      -lgee-0.8
+      -lglib-2.0
+      -lgobject-2.0
+      -lintl
+    ]
+    system ENV.cc, "test.c", "-o", "test", *flags
+    system "./test"
+  end
 end
+
+__END__
+diff --git a/gee/Makefile.in b/gee/Makefile.in
```

Comment:
We can probably do this with an `inreplace` rather than a patch, which is a bit nicer looking if we're going to have to carry it indefinitely.